resource-timing/xhr-resource-timing.html

```diff
@@ -0,0 +1,38 @@
+<!DOCTYPE HTML>
+<html>
+<head>
+<meta charset="utf-8" />
+<title>This test validates that a failed cross-origin fetch creates an opaque network timing entry.
+</title>
+<link rel="author" title="Noam Rosenthal" href="noam@webkit.org">
+<link rel="help" href="https://www.w3.org/TR/resource-timing-2/#sec-performanceresourcetiming"/>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script src="resources/entry-invariants.js"></script>
+<script src="/common/get-host-info.sub.js"></script>
+</head>
+<body>
+<script>
+const {REMOTE_ORIGIN} = get_host_info();
+
+async function test_xhr(url, event = "load") {
```

Comment:
I think it'd be nice (for more idiomatic tests and future reusability) to integrate this with resource-loaders.js and entry-invariants.js.

We already have a sync XHR loader: https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/web_tests/external/wpt/resource-timing/resources/resource-loaders.js;l=114?q=resource-loader&ss=chromium

Seems reasonable to add an async one that returns the load event timestamp as a value, and then verify that entry with a relevant invariant (and either baking in the load timestamp check into an invariant, or testing that separately).

Does that make sense?

Reply:
Sure, will get to it